### PR TITLE
Programatically sync the role with user list.

### PR DESCRIPTION
**Pull request by bkyryliuk**
Reviewers:
* @mistercrunch 
* @ascott 
* @vera-liu

### CODE REVIEW

File: tests/access_tests.py [hunk 1]
```diff
@@ -348,6 +348,50 @@ def test_request_access(self):
         gamma_user.roles.remove(sm.find_role('dummy_role'))
         session.commit()
 
+    def test_update_role(self):
```
Comment (ascott) [inline tests/access_tests.py]: is it typical in python to have one test that asserts multiple things? would it make sense to break this into multiple tests?

File: superset/views.py [hunk 2]
```diff
@@ -1073,6 +1073,27 @@ def msg(self):
 
 class Superset(BaseSupersetView):
     """The base views for Superset!"""
+
+    @has_access
+    @expose("/update_role/", methods=['POST'])
+    def update_role(self):
+        """Assigns a list of found users to the give role."""
```
Comment (ascott) [inline superset/views.py]: s/give/given

File: tests/access_tests.py (hunk 1 above)
Comment (mistercrunch) [inline tests/access_tests.py]: +1 on breaking tests down, refactoring shared parts out to methods

File: superset/views.py [hunk 3]
```diff
@@ -1073,6 +1073,27 @@ def msg(self):
...
+        role_name = data['role_name']
+        role = sm.find_role(role_name)
+        if not role:
+            return json_error_response(
+                __("%(role)s role was not found", role=role_name))
+        role.user = []
+        for user_email in user_emails:
+            user = sm.find_user(email=user_email)
+            if not user:
+                continue
```
Comment (mistercrunch) [inline superset/views.py]:
```
if user:
    role.user.append(user)
```

File: superset/views.py [hunk 4]
```diff
@@ -1073,6 +1073,27 @@ def msg(self):
...
+        """Assigns a list of found users to the give role."""
+        data = request.get_json(force=True)
+        user_emails = data['user_emails']
+        role_name = data['role_name']
+        role = sm.find_role(role_name)
+        if not role:
+            return json_error_response(
+                __("%(role)s role was not found", role=role_name))
+        role.user = []
+        for user_email in user_emails:
```
Comment (mistercrunch) [inline superset/views.py]: `for user_email in data['user_emails']` remove the need for 1082

File: superset/views.py [hunk 5]
```diff
@@ -1073,6 +1073,27 @@ def msg(self):
 
 class Superset(BaseSupersetView):
     """The base views for Superset!"""
+
+    @has_access
```
Comment (mistercrunch) [inline superset/views.py]: has_access_api

**Comment by mistercrunch** [comment]
LGTM once the build is fixed.

**Comment by bkyryliuk** [comment]
all done, thanks for the review @mistercrunch and @ascott  !

**Pull request merged**